src/services/poolTogether.js

```
@@ -0,0 +1,416 @@
+    const { feeFraction, openedBlock } = currentDraw;
+
+    const prizeIntervalMs = symbol === DAI ? 604800000 : 86400000; // DAI weekly, USDC daily
+    const { timestamp: blockTimestamp } = await provider.getBlock(openedBlock.toNumber());
+    drawDate = (blockTimestamp.toString() * 1000) + prizeIntervalMs; // adds 14 days to a timestamp in miliseconds
+
+    remainingTimeMs = drawDate - balanceTakenAt.getTime();
+    remainingTimeMs = remainingTimeMs < 0 ? 0 : remainingTimeMs;
+    const remainingTimeS = remainingTimeMs > 0 ? remainingTimeMs / 1000 : prizeIntervalMs / 1000;
+    const remainingBlocks = remainingTimeS / 15.0; // about 15 second block periods
+    const blockFixedPoint18 = utils.parseEther(remainingBlocks.toString());
+
+    const prizeSupplyRate = ptUtils.calculatePrizeSupplyRate(
+      supplyRatePerBlock,
+      feeFraction,
+    );
+
+    currentPrize = ptUtils.calculatePrize(
+      balance,
+      accountedBalance,
+      feeFraction,
+    );
+    prizeEstimate = ptUtils.calculatePrizeEstimate(
+      balance,
+      currentPrize,
+      blockFixedPoint18,
+      prizeSupplyRate,
+    );
+  }
+
+  return Promise.resolve({
+    currentPrize: formatMoney(utils.formatUnits(currentPrize.toString(), unitType)),
+    prizeEstimate: formatMoney(utils.formatUnits(prizeEstimate.toString(), unitType)),
+    drawDate,
+    remainingTimeMs,
+    totalPoolTicketsCount,
+    userInfo,
+  });
+}
+
+export const getSmartWalletTxFee = async (transaction: Object, useGasToken: boolean): Promise<TransactionFeeInfo> => {
+  const defaultResponse = { fee: new BigNumber(0) };
+  const estimateTransaction = {
+    data: transaction.data,
+    recipient: transaction.to,
+    value: transaction.amount,
+  };
+
+  const estimated = await smartWalletService
+    .estimateAccountTransaction(estimateTransaction)
+    .then(result => buildTxFeeInfo(result, useGasToken))
+    .catch(() => null);
```

Whilst we're returning null in this error state, please at least report the error to Sentry so we can capture a wider scope of issues.

---

Thanks, report added as well as error handling to this function 4bc3f3e

---

👍